libs/execution_context/ebpf_native.c

```diff
@@ -917,15 +922,16 @@ _ebpf_native_set_initial_map_values(_Inout_ ebpf_native_module_t* module)
 
             ebpf_handle_t handle_to_insert = ebpf_handle_invalid;
 
-            if (native_map_to_update->entry->definition.type == BPF_MAP_TYPE_ARRAY_OF_MAPS) {
+            ebpf_map_type_t map_type = native_map_to_update->entry->definition.type;
+            if (map_type == BPF_MAP_TYPE_ARRAY_OF_MAPS || map_type == BPF_MAP_TYPE_HASH_OF_MAPS) {
```

Comment:
This doesn't make sense. Only array type maps can be statically initialized as they have an implicit key value. Hash style maps have no implicit key value and can't be statically initialized.

Reply:
Analysis in https://github.com/microsoft/ebpf-for-windows/issues/2882#issuecomment-1730680224

Reply:
Thanks.  Agreed that the change in ebpf_native.c is not needed.
Then as discussed with @shankarseal, the **bpf2c** should flag the error that it cannot initialize the value of Hash-of-Maps. I will create a separate issue.


Create an issue to add the missing test cases.
- Hash of Maps with **ANY inner map type**, without static initializer.
- Array of Maps with **ANY inner map type**, with static initializer.